IdentityCore/src/cache/MSIDTokenCacheKey.m

```diff
@@ -169,4 +156,135 @@ + (NSString *)familyClientId:(NSString *)familyId
     return [NSString stringWithFormat:@"foci-%@", familyId];
 }
 
+- (BOOL)isEqualToTokenCacheKey:(MSIDTokenCacheKey *)key
+{
+    if (!key)
+    {
+        return NO;
+    }
+    
+    BOOL result = YES;
+    result &= (!self.account && !key.account) || [self.account isEqualToString:key.account];
+    result &= (!self.service && !key.service) || [self.service isEqualToString:key.service];
+    result &= (!self.type && !key.type) || [self.type isEqualToNumber:key.type];
```

Comment:
shouldn't authority, resource etc be checked here too?

Reply:
we will get rid of them after closing #54